src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,345 @@
+
+#include "gcs_actor_manager.h"
+#include <ray/common/ray_config.h>
+#include "gcs_actor_scheduling_strategy.h"
+#include "gcs_leased_worker.h"
+#include "gcs_node_manager.h"
+
+namespace ray {
+namespace gcs {
+
+ClientID GcsActor::GetNodeID() const {
+  return ClientID::FromBinary(actor_table_data_.address().raylet_id());
+}
+
+void GcsActor::SetNodeID(const ClientID &node_id) {
+  rpc::Address address(actor_table_data_.address());
+  address.set_raylet_id(node_id.Binary());
+  actor_table_data_.mutable_address()->CopyFrom(address);
+}
+
+void GcsActor::SetWorker(std::weak_ptr<GcsLeasedWorker> &&worker) {
+  worker_ = std::move(worker);
+  auto wk = worker_.lock();
+  RAY_CHECK(wk != nullptr);
+  actor_table_data_.mutable_address()->CopyFrom(wk->GetAddress());
+}
+
+std::shared_ptr<GcsLeasedWorker> GcsActor::GetWorker() const { return worker_.lock(); }
+
+WorkerID GcsActor::GetWorkerID() const {
+  auto worker = GetWorker();
+  return worker ? worker->GetWorkerID() : WorkerID::Nil();
+}
+
+void GcsActor::SetState(rpc::ActorTableData_ActorState state) {
+  actor_table_data_.set_state(state);
+}
+
+rpc::ActorTableData_ActorState GcsActor::GetState() const {
+  return actor_table_data_.state();
+}
+
+ActorID GcsActor::GetActorID() const {
+  const auto &task_spec = actor_table_data_.task_spec();
+  return ActorID::FromBinary(task_spec.actor_creation_task_spec().actor_id());
+}
+
+TaskSpecification GcsActor::GetTaskSpecification() const {
+  const auto &task_spec = actor_table_data_.task_spec();
+  return TaskSpecification(task_spec);
+}
+
+const rpc::ActorTableData &GcsActor::GetActorTableData() const {
+  return actor_table_data_;
+}
+
+rpc::ActorTableData *GcsActor::GetMutableActorTableData() { return &actor_table_data_; }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+GcsActorManager::GcsActorManager(boost::asio::io_context &io_context,
+                                 std::shared_ptr<gcs::RedisGcsClient> gcs_client,
+                                 std::weak_ptr<GcsNodeManager> &&gcs_node_manager)
+    : io_context_(io_context),
+      redis_gcs_client_(std::move(gcs_client)),
+      gcs_node_manager_(std::move(gcs_node_manager)),
+      gcs_actor_scheduling_strategy_(
+          new RandomGcsActorSchedulingStrategy(gcs_node_manager_.lock())) {
```

Comment:
Why would we ever want a specific strategy here? Even if we do, let's leave it out of this PR. The goal of this PR is to have the simplest possible implementation that still meets existing functionality, so let's not try to over-abstract here.

Reply:
We introduce this strategy because we want to do various special scheduling on GCS, such as making each node start only one actor, or scheduling some actors to the same node, and so on. Of course, there are also policy modules in raylet, but because the information synchronization of all nodes can not totally keep step with each other, it is difficult to choose a random raylet to do these complex customized policies, unless you always schedule policies to a specific raylet.

Anyway, I will remove the strategy abstraction for the time being to reduce the complexity of this PR.